### DISCUSSION TITLE
Truncation/Reset support

### DISCUSSION THREAD

**Discussion by wangandi** [MaterializeInc/materialize#7276] [Technical musings]
A number of sources or potential sources emit an event that calls for the deletion of all existing rows in previously in the source.

Examples:
* server sent sources can have reset events https://github.com/MaterializeInc/materialize/issues/2237#issuecomment-729259735
* Debezium has a truncate event. https://github.com/MaterializeInc/materialize/issues/6596
* Postgresql also has a truncate event. cc: @petrosagg

@frankmcsherry mentioned that differential is capable of supporting truncation.

**Comment by frankmcsherry** [comment]
I did?

Truncation in DD is usually "turn off dataflow and turn it on again". DD doesn't have an easy concept of that, unless you've retained an arrangement to explicitly retract.

**Reply by wangandi** [reply]
I definitely thought you mentioned you had some idea for truncation/reset support that we could hash out in greater detail this week. I could've been wrong on DD being the source of that support.
